### ISSUE TITLE
Cross-aarch64 example doesn't work - error: linker `cc` not found

### ISSUE THREAD

**Issue by hex0x0000** [oxalica/rust-overlay#47]
```
$ make test
nix-shell ./shell.nix --command \
	"cargo test --target aarch64-unknown-linux-gnu --no-run"
  Downloaded openssl-sys v0.9.60
  Downloaded cc v1.0.66
  Downloaded openssl v0.10.32
  Downloaded libc v0.2.85
  Downloaded 4 crates (821.3 KB) in 0.64s
   Compiling cc v1.0.66
   Compiling pkg-config v0.3.19
   Compiling autocfg v1.0.1
   Compiling libc v0.2.85
   Compiling bitflags v1.2.1
   Compiling foreign-types-shared v0.1.1
   Compiling openssl v0.10.32
   Compiling cfg-if v1.0.0
   Compiling lazy_static v1.4.0
   Compiling foreign-types v0.3.2
error: linker `cc` not found
  |
  = note: No such file or directory (os error 2)

error: aborting due to previous error

error: could not compile `openssl`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: linker `cc` not found
  |
  = note: No such file or directory (os error 2)

error: aborting due to previous error

error: linker `cc` not found
  |
  = note: No such file or directory (os error 2)

error: aborting due to previous error

error: build failed
make: *** [Makefile:8: all] Errore 101
```

NixOS version: 21.05

It doesn't work with my project either (which has the same files adapted to armv7):

shell.nix:
```nix
with import <nixpkgs>
{
  crossSystem = "armv7l-linux";
  overlays = [
    (import
      (builtins.fetchTarball
        "https://github.com/oxalica/rust-overlay/archive/master.tar.gz"))
  ];
};
mkShell {
  nativeBuildInputs = [
    buildPackages.rust-bin.stable.latest.minimal
    buildPackages.pkg-config
    pkgsBuildBuild.qemu
  ];
  buildInputs = [
    gcc
    openssl
  ];
}
```

.cargo/config:
```toml
[build]
target-dir = "target"

[target.armv7-unknown-linux-gnueabihf]
linker = "armv7l-unknown-linux-gnueabihf-gcc"
runner = "qemu-arm"
```

**Comment by oxalica** [comment]
In your cargo command (line 3), you choose `aarch64` target.
But in your `.cargo/config` and nix file, you use `armv7l`. They are totally different targets.

**Comment by hex0x0000** [comment]
@oxalica the files i posted (.cargo/config and shell.nix) come from another project. the error i posted is returned from [examples/cross-aarch64](https://github.com/oxalica/rust-overlay/tree/master/examples/cross-aarch64):

```
$ git clone https://github.com/oxalica/rust-overlay
$ cd rust-overlay/examples/cross-aarch64/
$ make all
nix-shell ./shell.nix --command \
	"cargo test --target aarch64-unknown-linux-gnu --no-run"
   Compiling pkg-config v0.3.19
   Compiling cc v1.0.66
   Compiling autocfg v1.0.1
   Compiling libc v0.2.85
   Compiling bitflags v1.2.1
   Compiling openssl v0.10.32
   Compiling foreign-types-shared v0.1.1
   Compiling cfg-if v1.0.0
   Compiling lazy_static v1.4.0
   Compiling foreign-types v0.3.2
error: linker `cc` not found
  |
  = note: No such file or directory (os error 2)

error: aborting due to previous error

error: could not compile `openssl`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: linker `cc` not found
  |
  = note: No such file or directory (os error 2)

error: aborting due to previous error

error: linker `cc` not found
  |
  = note: No such file or directory (os error 2)

error: aborting due to previous error

error: build failed
make: *** [Makefile:8: all] Errore 101
```

**Comment by oxalica** [comment]
I cannot reproduce your error even using `nix-shell --pure`.
What `nixpkgs` channel (the `<nixpkgs>`) are you using and what's its commit hash?

**Comment by hex0x0000** [comment]
`home-manager https://github.com/nix-community/home-manager/archive/release-21.05.tar.gz`

**Comment by oxalica** [comment]
I mean your `nixpkgs` (or maybe named `nixos`), not `home-manager`.

Please run this command and paste its output:
`nix-instantiate --eval -E 'builtins.readFile <nixpkgs/.git-revision>'`

**Comment by hex0x0000** [comment]
"110a2c9ebbf5d4a94486854f18a37a938cfacbbb"

**Comment by oxalica** [comment]
I still failed to reproduce the issue even on your nixpkgs revision.

Please check these things:
1. You don't have any of these file/directory exist: `~/.config/nixpkgs/overlays.nix` or `~/.config/nixpkgs/overlays`. If they exist, please temporarily move them to another place.
2. `examples/cross-aarch64/.cargo/config` is checkout-ed from repo `master` and is not modified.
3. Don't have any environment variable starting with `CARGO_`. You can check them by `env | grep "CARGO_"`. It should output nothing.

If you checked everything above but still fail to compile. Please try to run `nix-shell -p gcc` and then `make all` inside it.

**Comment by hex0x0000** [comment]
Apparently it works only with `nix-shell -p gcc`

**Comment by oxalica** [comment]
It should be fixed in c7ed82b2317b6b0d3d3a55ce4d12561f868858d0

Please checkout it and test again (should work without manual nix-shell).

**Comment by hex0x0000** [comment]
It works now! Thanks!
